core/src/event_id.rs

```diff
@@ -1,100 +1,70 @@
-/// last8Bytes(sha256(stream_controller_DID)), // 16 bytes
-/// last4Bytes(init_event_CID) // 8 bytes
-/// cbor(eventHeight), // 1-3 bytes
-/// eventCID // 36 bytes
-///   0x01 cidv1, 1 byte
-///   0x71 dag-cbor, 1 byte
-///   0x12 sha2-256, 1byte
-///   0x20 varint(hash length), 1 byte
-///   hash bytes, 32 bytes
-
-/// Network values from https://cips.ceramic.network/tables/networkIds.csv
-/// Ceramic Pubsub Topic, Timestamp Authority
-#[derive(Debug)]
-pub enum Network {
-    /// /ceramic/mainnet Ethereum Mainnet (EIP155:1)
-    Mainnet,
-
-    /// /ceramic/testnet-clay Ethereum Gnosis Chain
-    TestnetClay,
 
-    /// /ceramic/dev-unstable Ethereum Gnosis Chain
-    DevUnstable,
-
-    /// /ceramic/local-$(randomNumber) Ethereum by Truffle Ganache
-    Local(u32),
-
-    /// None
-    InMemory,
-}
-
-impl Network {
-    /// network.to_u64() to get the network as a u64
-    pub fn to_u64(&self) -> u64 {
-        match self {
-            // https://github.com/ceramicnetwork/CIPs/blob/main/tables/networkIds.csv
-            Network::Mainnet => 0x00,
-            Network::TestnetClay => 0x01,
-            Network::DevUnstable => 0x02,
-            Network::InMemory => 0xff,
-            Network::Local(id) => 0x01_0000_0000_u64 + u64::from(*id),
-        }
-    }
-}
+use crate::network::Network;
 
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
 /// EventId is the event data as a recon key
 pub struct EventId(#[serde(with = "serde_bytes")] Vec<u8>);
 
 impl EventId {
+    /// Create a builder for constructing EventIds.
+    pub fn builder() -> Builder<Init> {
+        Builder { state: Init }
+    }
     /// EventId.new builds a Vec<u8> with the event id data.
     pub fn new(
-        network_id: Network,
-        separator: &str,
+        network: &Network,
+        sort_value: &str,
         controller: &str,
         init: &Cid,
         event_height: u64,
         event_cid: &Cid,
     ) -> EventId {
-        let mut event_height_cbor = Encoder::from_memory();
-        event_height_cbor.encode([event_height]).unwrap();
-        EventId(
-            [
-                varint(0xce, &mut [0_u8; 10]),                // streamid varint
-                varint(0x05, &mut [0_u8; 10]),                // cip-124 EventID varint
-                varint(network_id.to_u64(), &mut [0_u8; 10]), // network_id varint
-                last8_bytes(&sha256_digest(separator)),       // separator [u8; 8]
-                last8_bytes(&sha256_digest(controller)),      // controller [u8; 8]
-                last4_bytes(init.to_bytes().as_slice()),      // StreamID [u8; 4]
-                event_height_cbor.as_bytes(),                 // event_height cbor unsigned int
-                //varint(event_height, &mut [0_u8; 10]), // event_height varint
-                event_cid.to_bytes().as_slice(), // [u8]
-            ]
-            .concat(),
-        )
+        EventId::builder()
```

Comment:
May be easier to just add my functon to parce out the CID here rather than creating a merge conflict 
```rust
use unsigned_varint::{decode::u64 as de_varint, encode::u64 as varint};

/// try to parse a CID out of a CIP-124 EventID
    pub fn cid(&self) -> Option<Cid> {
        let (streamid, remainder) = de_varint(&self.0).unwrap_or_default();
        if streamid != 0xce {
            return None; // not a streamid
        };

        let (typeid, remainder) = de_varint(remainder).unwrap_or_default();
        if typeid != 0x05 {
            return None; // not a CIP-124 EventID
        };

        let (_network_id, mut remainder) = de_varint(remainder).unwrap_or_default();

        // strip separator [u8; 8] controller [u8; 8] StreamID [u8; 4]
        remainder = &remainder[(8 + 8 + 4)..];

        // hight cbor unsigned integer
        if remainder[0] <= 23 {
            // 0 - 23
            remainder = &remainder[1..]
        } else if remainder[0] == 24 {
            // u8
            remainder = &remainder[2..]
        } else if remainder[0] == 25 {
            // u16
            remainder = &remainder[3..]
        } else if remainder[0] == 26 {
            // u32
            remainder = &remainder[5..]
        } else if remainder[0] == 27 {
            // u64
            remainder = &remainder[9..]
        } else {
            // not a cbor unsigned int
            return None;
        };
        match Cid::read_bytes(remainder) {
            Ok(v) => Some(v),
            Err(_) => None, // not a CID
        }
    }
}
```

Reply:
This looks great, I'll just add it to my PR.